core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -426,7 +427,10 @@ default void setResponseTimeout(Duration responseTimeout) {
      * Returns {@link Response} timeout handler which is executed when
      * the {@link Response} is not completely received within the allowed {@link #responseTimeoutMillis()}
      * or the default {@link ClientOptions#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * @deprecated Use {@link #whenResponseTimingOut()} or {@link #whenResponseTimedOut()}
      */
+    @Deprecated
     @Nullable
     Runnable responseTimeoutHandler();
```

Comment:
Could return `null` by default and delete the old implementations?

Reply:
That sounds good!